File: solutions/3347-distribute-elements-into-two-arrays-i/distribute-elements-into-two-arrays-i.cs
public class Solution {
    public int[] ResultArray(int[] nums) {
        int n = nums.Length;
        int[] arr = new int[n];
        arr[0] = nums[0];
        arr[n - 1] = nums[1];
        int idx = 0, revIdx = n - 1;
        for (int i = 2; i < n; i++) {
            if (arr[idx] > arr[revIdx]) {
                arr[++idx] = nums[i];
            } else {
                arr[--revIdx] = nums[i];
            }
        }
        Array.Reverse(arr, revIdx, n - revIdx);
        return arr;
    }
}

File: solutions/3347-distribute-elements-into-two-arrays-i/distribute-elements-into-two-arrays-i.rs
impl Solution {
    pub fn result_array(nums: Vec<i32>) -> Vec<i32> {
        let n = nums.len();
        let mut arr = vec![0; n];
        arr[0] = nums[0];
        arr[n - 1] = nums[1];
        let mut idx = 0;
        let mut rev_idx = n - 1;
        for i in 2..n {
            if arr[idx] > arr[rev_idx] {
                idx += 1;
                arr[idx] = nums[i];
            } else {
                rev_idx -= 1;
                arr[rev_idx] = nums[i];
            }
        }
        arr[rev_idx..].reverse();
        arr
    }
}